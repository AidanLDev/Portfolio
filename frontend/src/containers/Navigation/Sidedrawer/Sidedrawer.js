import React from 'react';
import Backdrop from 'components/Backdrop/Backdrop';
import Button from 'components/Button/Button';
import Logo from 'components/Logo/Logo';
import Navigation from 'containers/Navigation/Navigation';

import styles from './Style.scss';

const sidedrawer = props => {
  return (
    <div>
      <Backdrop show={props.showDrawer} clicked={props.closeDrawer}/>
      <div className={styles.sidedrawer}>
        <Logo />
        <hr/>
        {/* Logo */}
        <nav>
          <Navigation />
        </nav>
        <hr/>
        <Button clicked={props.closeDrawer}>Close Menu</Button>
      </div>
    </div>
  );
}

export default sidedrawer;
