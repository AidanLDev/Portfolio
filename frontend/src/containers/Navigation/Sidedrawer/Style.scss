@import "pallet";
@import "variables";
@import "mixins";

.sidedrawer {
  position: fixed;
  width: 280px;
  max-width: 70%;
  height: 100%;
  left: 0;
  top: 0;
  z-index: 200;
  background-color: $modalBackground;
  padding: 32px 0 0 /*  16px  */;
  box-sizing: border-box;
  transition: transform 1.3s ease-out;

  button {
    bottom: 20%;
    position: absolute;
    left: 31%;
  }
}

@media (min-width: 500px) {
  .sidedrawer {
    display: none;
  }
}

.open {
  transform: translateX(0);
}

.close {
  transform: translateX(-100%);
}
