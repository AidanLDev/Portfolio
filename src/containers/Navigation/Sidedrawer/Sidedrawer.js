import React from 'react';
import Button from '@material-ui/core/Button';

import Backdrop from 'components/Backdrop/Backdrop';
import Logo from 'components/Logo/Logo';
import Navigation from 'containers/Navigation/Navigation';

import styles from './Style.module.scss';

const sidedrawer = (props) => {
  return (
    <div>
      <Backdrop show={props.showDrawer} clicked={props.closeDrawer} />
      <div className={styles.sidedrawer}>
        <Logo />
        <hr />
        {/* Logo */}
        <nav>
          <Navigation />
        </nav>
        <hr />
        <Button onClick={props.closeDrawer}>Close Menu</Button>
      </div>
    </div>
  );
};

export default sidedrawer;
