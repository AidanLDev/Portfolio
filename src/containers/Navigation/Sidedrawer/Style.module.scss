@import"styles/pallet.scss";
@import"styles/variables.scss";
@import"styles/mixins.scss";

.sidedrawer {
  position: fixed;
  width: 280px;
  max-width: 70%;
  height: 100%;
  left: 0;
  top: 0;
  z-index: 200;
  background-color: $modalBackground;
  padding: 32px 0 0 /*  16px  */;
  box-sizing: border-box;
  transition: transform 1.3s ease-out;

  button {
    color: $lightGrey;
    bottom: 20%;
    position: absolute;
    left: 31%;
  }
}

@media (min-width: 501px) {
  .sidedrawer {
    display: none;
  }
}

.open {
  transform: translateX(0);
}

.close {
  transform: translateX(-100%);
}
