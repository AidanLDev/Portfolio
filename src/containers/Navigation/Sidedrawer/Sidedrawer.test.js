import React from 'react';
import Sidedrawer from './Sidedrawer';
import { shallow } from 'enzyme';

it('Renders Sidedrawer Component Correctly', () => {
  const component = shallow(<Sidedrawer />);
  expect(component).toMatchSnapshot();
});
