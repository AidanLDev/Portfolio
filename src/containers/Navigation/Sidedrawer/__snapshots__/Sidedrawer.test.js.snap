// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Sidedrawer Component Correctly 1`] = `
<div>
  <backdrop />
  <div
    className="sidedrawer"
  >
    <logo />
    <hr />
    <nav>
      <Navigation />
    </nav>
    <hr />
    <WithStyles(ForwardRef(Button))>
      Close Menu
    </WithStyles(ForwardRef(Button))>
  </div>
</div>
`;
